apollo-normalized-cache/src/commonMain/kotlin/com/apollographql/apollo3/cache/normalized/ClientCacheExtensions.kt

```diff
@@ -150,13 +148,20 @@ fun <D : Query.Data> ApolloCall<D>.watch(
             .watch(lastResponse?.data) { _, _ ->
               // If the exception is ignored (refetchThrows is false), we should continue watching - so retry
               !refetchThrows
-            }.onStart {
-              if (lastResponse != null) {
-                emit(lastResponse!!)
-              }
             }
     )
   }
+
+  /**
+   * We buffer the returned flow so that the subscription to the store happens in the same stacktrace as the last element is
+   * received and *before* the last element is emitted downstream.
+   *
+   * This allows callers to use the last element as a synchronisation point to modify the store and still have the watcher
```

Comment:
👏

Reply:
Well, this broke the stress test... I'm assuming because when the coroutine run on different threads, just relying on the callstack is not enough as coroutine #1 can be interrupted any time...